application/common/models/User.php

```diff
@@ -753,19 +799,16 @@ public function isPromptForMfa(): bool
     }
 
     /**
-     * Update the first nag_for_*_after date that has passed
+     * Update the nag_for_*_after date that corresponds to the current nag state
      */
     public function updateNagDates()
     {
-        // save current time to avoid race condition
-        $now = time();
-
-        if (strtotime($this->nag_for_mfa_after) < $now) {
+        if ($this->getNagState() === self::NAG_ADD_MFA || $this->getNagState() === self::NAG_REVIEW_MFA) {
```

Comment:
I think we'll want the interval between nags-to-set-up-MFA to be different than the interval between nags-to-review-your-MFA. For example, if I haven't set up MFA, I could understand being nagged every couple of months. However, if I have set up MFA, I wouldn't want to be nagged about reviewing it more often than every 6 - 12 months.

Reply:
Done.